services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -481,6 +474,60 @@ public LinkedAccountsResource linkedAccounts() {
         return new LinkedAccountsResource(session, request, client, auth, event, user);
     }
 
+    @Path("/applications")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public Response applications() {
+        checkAccountApiEnabled();
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_APPLICATIONS);
+
+        List<ClientModel> clients = new LinkedList<>();
```

Comment:
Just a suggestion for a possible code clarity enhancement (I don't see it as an issue and I'll approve regardless). Maybe I'm wrong, but couldn't we use a `Set` (or `SortedSet ` to retain order) here instead of `List` to get rid of many conditions like `if(!clients.contains(client))`? It might make the code cleaner.

Reply:
Done